src/components/ui4/ModalUi4.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react'
+import { BackHandler, Dimensions, StyleSheet, TouchableWithoutFeedback } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import { Gesture, GestureDetector } from 'react-native-gesture-handler'
+import { cacheStyles } from 'react-native-patina'
+import Animated, { runOnJS, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
+import { BlurView } from 'rn-id-blurview'
+
+import { fixSides, mapSides, sidesToPadding } from '../../util/sides'
+import { Theme, useTheme } from '../services/ThemeContext'
+
+export interface ModalPropsUi4<T = unknown> {
+  bridge: AirshipBridge<T>
+  children?: React.ReactNode
+
+  // Called when the user taps outside the modal or clicks the back button:
+  onCancel: () => void
+
+  // Internal padding to place inside the component.
+  paddingRem?: number[] | number
+
+  // Gives the box a border:
+  warning?: boolean
+}
+
+const safeAreaGap = 64 // Overkill to avoid bottom of screen
+const duration = 300
+
+/**
+ * A modal that slides a modal up from the bottom of the screen
+ * and dims the rest of the app.
+ */
+export function ModalUi4<T>(props: ModalPropsUi4<T>): JSX.Element {
+  const { bridge, children, onCancel, paddingRem, warning = false } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const padding = sidesToPadding(mapSides(fixSides(paddingRem, 1), theme.rem))
+  const closeThreshold = theme.rem(6)
+  const dragSlop = theme.rem(1)
+
+  //
+  // Shared state
+  //
+
+  const opacity = useSharedValue(0)
+  const offset = useSharedValue(Dimensions.get('window').height)
+
+  //
+  // Effects
+  //
+
+  React.useEffect(() => bridge.on('clear', onCancel), [bridge, onCancel])
+
+  React.useEffect(() => {
+    // Animate in:
+    opacity.value = withTiming(1, { duration })
+    offset.value = withTiming(0, { duration })
+
+    // Animate out:
+    bridge.on('result', () => {
+      opacity.value = withTiming(0, { duration })
+      offset.value = withTiming(Dimensions.get('window').height, { duration }, () => runOnJS(bridge.remove)())
+    })
+  }, [bridge, opacity, offset])
+
+  React.useEffect(() => {
+    const backHandler = BackHandler.addEventListener('hardwareBackPress', () => {
+      onCancel()
+      return true
+    })
+    return () => backHandler.remove()
+  }, [onCancel])
+
+  const gesture = Gesture.Pan()
```

Comment:
Not within the scope of this PR, but scrollable areas conflict with modal
drag down gestures.

Propose we design:
1. More obvious scrollable area bounds within the modal and/or
2. An obvious reserved spot for modal gesture activation, some horizontal drag
   bar icon
3. OR Something fancy that dynamically addresses this: https://raw.githubusercontent.com/gorhom/react-native-bottom-sheet/HEAD/preview.gif